drivers/display/display_sdl.c

```diff
@@ -49,32 +77,42 @@ static inline uint32_t mono_pixel_order(uint32_t order)
 	}
 }
 
-static int sdl_display_init(const struct device *dev)
+static void exec_sdl_task(const struct device *dev, const struct sdl_display_task *task)
 {
-	const struct sdl_display_config *config = dev->config;
 	struct sdl_display_data *disp_data = dev->data;
-	bool use_accelerator = true;
-	LOG_DBG("Initializing display driver");
 
-	IF_DISABLED(CONFIG_SDL_DISPLAY_USE_HARDWARE_ACCELERATOR, (use_accelerator = false));
+	K_SPINLOCK(&disp_data->lock) {
```

Comment:
Minor:
* native_sim is not SMP capable (and by design it won't be) => this is just a `irq_lock()`
* In native_sim we purposely do not allow asynchronous mechanisms to generate interrupts into Zephyr, which means no other Zephyr thread will call into exec_sdl_task() while this is executing even with interrupts unlocked.

=> this lock is not needed, but the harm of having is just more complex code and a bit of overhead. I'm  not rejecting the PR due to this.

This also applies to all other `K_SPINLOCK`s added in this PR

Reply:
Good point, I removed the spinlock.